### DISCUSSION TITLE
How should I change landscape or portrait orientation mode?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13547] [Q&A] [answered]
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/96ebc93d-c657-4450-9550-3f0a3d895c92)

As we know, running programs in mobile phones there are two orientation modes: landscape or portrait.

In the UI of my program, either the layout or style between landscape mode and portrait differ a lot.

I want to change the orientation by a button (not by the gravity sensor of the phone) when the program is running (without restart the program)

 How can I achieve this?

**Comment by timunie** [comment, marked as answer]
different options... 

first of all i'd have a bool in my MainViewModel indicating orientation I whish. Then you can:
- Have two UserControls and bind to IsVisible
- Use a behavior
- Use a converter
- Bind to Style-classes ...

**Reply by kyuranger** [reply]
The two usercontrols ways sound simple and convenient. But I'm not sure Avalonia will handle and render those invisible usercontrols which will cause longer rendering times.

**Reply by timunie** [reply]
No, hidden controls don't add much overhead. These have a size of 0x0 px

**Reply by thevortexcloud** [reply]
From memory, having two different views/layouts is also how Android handles it using their native UI. It does however have the side effect that if you need to support rotating the view on the fly, then you need to keep both views in sync/reload all the data back into the view. Which can be somewhat painful.

They mention some work arounds here which the general principle is just as applicable to Avalonia.

https://developer.android.com/guide/topics/resources/runtime-changes

But as long as you are using MVVM this should not be too bad in most instances. If you are storing some kind of state in code behind it can get tricky,

**Reply by kyuranger** [reply]
Thanks for your suggestion.
